### PR TITLE
Document CallCredentials CancellationToken

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/grpc/grpc-dotnet/issues/2108

Ready for review. Shouldn't be merged until gRPC 2.54.0 is released.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/grpc/authn-and-authz.md](https://github.com/dotnet/AspNetCore.Docs/blob/5b58b4cdc85cded886dd5c36f3865d14f0b8d62f/aspnetcore/grpc/authn-and-authz.md) | [Authentication and authorization in gRPC for ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/authn-and-authz?branch=pr-en-us-29185) |

### CODE REVIEW

**Review by wadepickett — APPROVED** [review]
Looks good to me, thanks James.

**Comment by wadepickett** [comment]
@JamesNK,  I can look at moving the inline code to an app sample at some point later, likely after the rush of 8.0 settles down.  Do we happen to have sample app already someplace for this?  Either way, I can create one, but good to know.

**Comment by wadepickett** [comment]
Reminder for me: "Shouldn't be merged until gRPC 2.54.0 is released."

File: aspnetcore/grpc/authn-and-authz.md [hunk 1]
```diff
@@ -334,26 +338,29 @@ public bool DoAuthenticatedCall(
...
 ```
 
+#### Set the bearer token with `CallCredentials`
+
 Configuring `ChannelCredentials` on a channel is an alternative way to send the token to the service with gRPC calls. A `ChannelCredentials` can include `CallCredentials`, which provide a way to automatically set `Metadata`.
 
-`CallCredentials` is run each time a gRPC call is made, which avoids the need to write code in multiple places to pass the token yourself. Note that `CallCredentials` are only applied if the channel is secured with TLS. `CallCredentials` aren't applied on unsecured non-TLS channels.
+Benefits of using `CallCredentials`:
+
+* Authentication is centrally configured on the channel. The token doesn't need to be manually provided to gRPC call.
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/authn-and-authz.md:347]:
```suggestion
* Authentication is centrally configured on the channel. The token doesn't need to be manually provided to the gRPC call.
```
Although yours is correct, I prefer using the definite article

File: aspnetcore/grpc/authn-and-authz.md [hunk 2]
```diff
@@ -334,26 +338,29 @@ public bool DoAuthenticatedCall(
...
+* The `CallCredentials.FromInterceptor` callback is asynchronous. Call credentials can fetch a credential token from an external system if required. Async calls inside the callback should use the `CancellationToken` on `AuthInterceptorContext`.
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/authn-and-authz.md:348]:
```suggestion
* The `CallCredentials.FromInterceptor` callback is asynchronous. Call credentials can fetch a credential token from an external system if required. Asynchronous calls inside the callback should use the `CancellationToken` on `AuthInterceptorContext`.
```

**Review by Rick-Anderson — APPROVED** [review]

File: aspnetcore/grpc/authn-and-authz.md [hunk 3]
```diff
@@ -69,23 +69,27 @@ public bool DoAuthenticatedCall(
...
 ```
 
-Configuring `ChannelCredentials` on a channel is an alternative way to send the token to the service with gRPC calls. A `ChannelCredentials` can include `CallCredentials`, which provide a way to automatically set `Metadata`. `CallCredentials` is run each time a gRPC call is made, which avoids the need to write code in multiple places to pass the token yourself.
+#### Set the bearer token with `CallCredentials`
+
+Configuring `ChannelCredentials` on a channel is an alternative way to send the token to the service with gRPC calls. A `ChannelCredentials` can include `CallCredentials`, which provide a way to automatically set `Metadata`.
+
+Benefits of using `CallCredentials`:
+
+* Authentication is centrally configured on the channel. The token doesn't need to be manually provided to gRPC call.
```
Comment (JamesNK) [inline aspnetcore/grpc/authn-and-authz.md:78]:
```suggestion
* Authentication is centrally configured on the channel. The token doesn't need to be manually provided to the gRPC call.
```

File: aspnetcore/grpc/authn-and-authz.md [hunk 4]
```diff
@@ -69,23 +69,27 @@ public bool DoAuthenticatedCall(
...
+* The `CallCredentials.FromInterceptor` callback is asynchronous. Call credentials can fetch a credential token from an external system if required. Async calls inside the callback should use the `CancellationToken` on `AuthInterceptorContext`.
```
Comment (JamesNK) [inline aspnetcore/grpc/authn-and-authz.md:79]:
```suggestion
* The `CallCredentials.FromInterceptor` callback is asynchronous. Call credentials can fetch a credential token from an external system if required. Asynchronous calls inside the callback should use the `CancellationToken` on `AuthInterceptorContext`.
```

**Comment by JamesNK** [comment]
2.54.0 is out now - https://github.com/grpc/grpc-dotnet/releases/tag/v2.54.0

**Pull request merged**